src/commands/runDraftTool/runDraftIngress.ts

```diff
@@ -0,0 +1,749 @@
+         sort(getAsyncOptions(certs, certToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'Certificate',
+            placeHolder: 'Certificate',
+            noPicksMessage: 'No certificates found'
+         }
+      );
+
+      wizardContext.certificate = (await certs).find(
+         (cert) => certToItem(cert).label === certPick.label
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return !wizardContext.newSSLCert;
+   }
+}
+
+class PromptAksCluster extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.aksSubscription === undefined) {
+         throw Error('AKS Subscription is undefined');
+      }
+      if (wizardContext.aksResourceGroup === undefined) {
+         throw Error('AKS Resource Group is undefined');
+      }
+
+      const clusters = getAysncResult(
+         this.az.listAksClusters(
+            wizardContext.aksSubscription,
+            wizardContext.aksResourceGroup
+         )
+      );
+      const clusterToItem = (cluster: ManagedClusterItem) => ({
+         label: cluster.managedCluster.name || ''
+      });
+      const clusterPick = await wizardContext.ui.showQuickPick(
+         sort(getAsyncOptions(clusters, clusterToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'AKS Cluster',
+            placeHolder: 'AKS Cluster',
+            noPicksMessage: 'No AKS Clusters found'
+         }
+      );
+
+      wizardContext.aks = (await clusters).find(
+         (cluster) => clusterToItem(cluster).label === clusterPick.label
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class ExecuteCreateCertificate extends AzureWizardExecuteStep<WizardContext> {
+   public priority: number = 1;
+
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async execute(
+      wizardContext: WizardContext,
+      progress: vscode.Progress<{
+         message?: string | undefined;
+         increment?: number | undefined;
+      }>
+   ): Promise<void> {
+      const kv = wizardContext.kv;
+      if (typeof kv === 'undefined') {
+         throw Error('Key vault is undefined');
+      }
+      const dns = wizardContext.dns;
+      if (typeof dns === 'undefined') {
+         throw Error('DNS is undefined');
+      }
+      const name = wizardContext.certificateName;
+      if (typeof name === 'undefined') {
+         throw Error('Certificate name is undefined');
+      }
+
+      const pkiServerAuth = '1.3.6.1.5.5.7.3.1'; // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-ppsec/651a90f3-e1f5-4087-8503-40d804429a88
+      progress.report({message: 'Creating certificate'});
+      const resp = await this.az.createCertificate(kv, name, {
+         issuerName: 'Self',
```

Comment:
I may be missing something but how come we're using self-signed certificates?

Reply:
They have the option of selecting a cert or generating a self-signed one so they can (and should for prod) select a CA signed cert. 

Generating a signed cert here requires a lot more thought. Key Vault has [limited options](https://learn.microsoft.com/en-us/azure/key-vault/certificates/create-certificate-signing-request?tabs=azure-portal) on how we can integrate with a CA. In my opinion it's best to let the user handle creating / managing the cert because of how many options there are to do that (there's no one size fits all solution). There's so many CAs, expiration dates matter, which cli tool do we bundle, how many options do we give the user vs letting them choose themselves are all considerations. We can potentially do that in future iterations but it just would required lots of discussion + decisions.

Letting them pick a cert and warning the user that self-signed shouldn't be used for prod seems like the way to go for now.

Reply:
that makes sense!